codegen/src/lib.rs

```diff
@@ -409,11 +409,18 @@ fn default_substitutes(crate_path: &syn::Path) -> TypeSubstitutes {
         // `EncodeAsType` the bytes would be re-encoded. This leads to the bytes
         // being altered by adding the length prefix in front of them.
 
-        // Note: Not sure if this is appropriate or not. The most recent polkadot.rs file does not have these.
+        // This allows things calls that require an `UncheckedExtrinsic` type to be given our
+        // util one, which was be converted from a Vec<u8>:
         (
             parse_quote!(sp_runtime::generic::unchecked_extrinsic::UncheckedExtrinsic),
             parse_quote!(#crate_path::utils::UncheckedExtrinsic),
         ),
+        // This extends the above to also work with runtimes that set the UncheckedExtrinsic
+        // type to the one from pallet_revive (used for handling EVM transactions):
+        (
+            parse_quote!(pallet_revive::evm::runtime::UncheckedExtrinsic<Address, Call>),
```

Comment:
I wonder whether we need something special for [Acala uncheckedExtrinsic](https://github.com/AcalaNetwork/Acala/blob/master/primitives/src/unchecked_extrinsic.rs#L44-#L47) as well and double-check other chains as well?

Reply:
Yeah it looks like Subxt would run into issues with that type too when decoding from V14 metadata offhand!

That type doesn't expose _any_ of the generics we need though, so there's no obvious way to make it work for V14 metadata without somewhat custom hacks!

The second thing of being able to encode bytes into that type would be easy enough for them to create a type substitution for.

Reply:
It's not big issue anyway since it only concern fetching metadata v14 and subxt by default is using v15